app/org/sagebionetworks/bridge/services/backfill/StudyIdBackfill.java

```diff
@@ -63,15 +69,19 @@ void doBackfill(final BackfillTask task, final BackfillCallback callback) {
                         try {
                             String healthCode = healthId.getCode();
                             if (healthCode != null) {
-                                boolean set = healthCodeDao.setStudyId(healthCode, study.getIdentifier());
-                                if (set) {
-                                    callback.newRecords(createRecord(task, study, account, "backfilled"));
+                                final String studyId = healthCodeDao.getStudyIdentifier(healthCode);
+                                if (isBlank(studyId)) {
+                                    String msg = "Backfill needed as study ID is blank.";
+                                    callback.newRecords(backfillRecordFactory.createOnly(task, study, account, msg));
+                                } else {
+                                    String msg = "Study ID already exists.";
+                                    callback.newRecords(backfillRecordFactory.createOnly(task, study, account, msg));
```

Comment:
Would it be cleaner to just delete this class, since the study ID backfill is done and the code is being deleted at the dao level?